x-pack/plugins/infra/public/utils/header_action_menu_provider.tsx

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { AppMountParameters } from 'kibana/public';
+
+interface ContextProps {
+  setHeaderActionMenu?: AppMountParameters['setHeaderActionMenu'];
+}
+
+export const HeaderActionMenuContext = React.createContext<ContextProps>({});
+
+export const HeaderActionMenuProvider: React.FC<Required<ContextProps>> = ({
+  setHeaderActionMenu,
+  children,
+}) => {
+  return (
+    <HeaderActionMenuContext.Provider value={{ setHeaderActionMenu }}>
+      {children}
+    </HeaderActionMenuContext.Provider>
+  );
+};
```

Comment:
It's a way to abstract the `value` away into a meaningful name.

I think moving forward it would be interesting to add an `InfraContext` that holds all of these little things instead of having 3-4 wrappers, but that's a story for another PR.